### PR TITLE
Validate manifest file

### LINKED ISSUE #6050: Add PCLI command to validate the recovery manifest file
Create a new PCLI command to validate the contents of the emergency recovery manifest file. It should provide details about what is missed and/or invalid.

### PR DESCRIPTION
**Pull request by lpetrovic05**
closes #6050

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/ValidateManifestFileCommand.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        validateFieldExists(file.recovery().state().timestamp(), "recovery->state->timestamp");
+        validateFieldExists(file.recovery().boostrap(), "recovery->boostrap");
+        validateFieldExists(file.recovery().boostrap().timestamp(), "recovery->boostrap->timestamp");
+        validateFieldExists(file.recovery().pkg(), "recovery->package");
+        validateFieldExists(file.recovery().pkg().locations(), "recovery->package->locations");
+        if (file.recovery().pkg().locations().size() == 0) {
+            throw new IOException(
+                    "The file should have at least one location in the recovery->package->locations field");
+        }
+        List<Location> locations = file.recovery().pkg().locations();
```
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/ValidateManifestFileCommand.java:65]:
```suggestion
        final List<Location> locations = file.recovery().pkg().locations();
```

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/ValidateManifestFileCommand.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        }
+        validateFieldExists(file.recovery(), "recovery");
+        validateFieldExists(file.recovery().state(), "recovery->state");
+        validateFieldExists(file.recovery().state().hash(), "recovery->state->hash");
+        validateFieldExists(file.recovery().state().timestamp(), "recovery->state->timestamp");
+        validateFieldExists(file.recovery().boostrap(), "recovery->boostrap");
+        validateFieldExists(file.recovery().boostrap().timestamp(), "recovery->boostrap->timestamp");
+        validateFieldExists(file.recovery().pkg(), "recovery->package");
+        validateFieldExists(file.recovery().pkg().locations(), "recovery->package->locations");
+        if (file.recovery().pkg().locations().size() == 0) {
```
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/ValidateManifestFileCommand.java:61]: Not sure the data type of the object returned by `.locations()`, but if it is a collection then you do do the following:
```suggestion
        if (file.recovery().pkg().locations().isEmpty()) {
```

**Review by cody-littley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/emergencyfile/Recovery.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * @param state
+ * 		information about the state written to disk
+ * @param boostrap
+ * 		information about the state used to bootstrap event recovery. Not written during normal
+ * 		operation. Only written during event recovery.
+ * @param pkg information about where to find the emergency recovery package
+ * @param stream information about the various file streams
+ */
+public record Recovery(State state, Boostrap boostrap, @JsonProperty("package") Package pkg, Stream stream) {}
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/emergencyfile/Recovery.java:32]: Nullability markings on all records.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/emergencyfile/Boostrap.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+ * Data about the bootstrap state loaded during event recovery (the starting state)
+ *
+ * @param timestamp
+ * 		the consensus timestamp of the bootstrap state
+ */
+public record Boostrap(
+        @JsonSerialize(using = ToStringSerializer.class) @JsonDeserialize(using = InstantDeserializer.class)
+                Instant timestamp) {}
```
[the remark this answers is not in the document]
Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/emergencyfile/Boostrap.java:33]: done

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/emergencyfile/Location.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import java.net.URL;
+
+/**
+ * A location where a recovery package can be downloaded from
+ * @param type the type of package file (e.g. "zip")
+ * @param url the URL where the package can be downloaded from
+ * @param hash the hash of the package file
+ */
+public record Location(
+        String type,
```
[the remark this answers is not in the document]
Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/emergencyfile/Location.java:33]: done

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/ValidateManifestFileCommand.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    @SuppressWarnings("unused")
+    @CommandLine.Parameters(
+            description = "the path to dir containing manifest file which should be named emergencyRecovery.yaml")
+    private void setDir(final Path dir) {
+        this.pathMustExist(dir);
+        this.dir = dir;
+    }
+
+    @Override
+    public Integer call() throws IOException {
```
[the remark this answers is not in the document]
Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/ValidateManifestFileCommand.java:48]: done

**Review by JeffreyDallas — APPROVED** [review]

**Review by qnswirlds — APPROVED** [review]

**Comment by lpetrovic05** [comment]
replaced by #6842

**Pull request closed without merging**